File: backend/src/test/java/lan/dk/podcastserver/service/ByteRangeResourceHandlerTest.java
package lan.dk.podcastserver.service;

import org.junit.Test;
import org.springframework.core.io.Resource;

import javax.servlet.http.HttpServletRequest;

import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Paths;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

/**
 * Created by kevin on 02/09/2017
 */
public class ByteRangeResourceHandlerTest {

    private ByteRangeResourceHandler brh = new ByteRangeResourceHandler();

    @Test
    public void should_expose_a_file_resource() throws IOException {
        /* GIVEN */
        Path file = Paths.get("/tmp/foo");
        HttpServletRequest request = mock(HttpServletRequest.class);
        when(request.getAttribute(ByteRangeResourceHandler.ATTR_FILE)).thenReturn(file);

        /* WHEN  */
        Resource r = brh.getResource(request);

        /* THEN  */
        assertThat(r.getFile().toPath()).isEqualByComparingTo(file);
    }

    @Test
    public void should_raise_exception_if_no_attribute() {
        /* GIVEN */
        HttpServletRequest request = mock(HttpServletRequest.class);
        /* WHEN  */
        assertThatThrownBy(() -> brh.getResource(request))
        /* THEN  */
                .hasMessage("Error during serving of byte range resources");
    }

}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/service/ByteRangeResourceHandlerTest.kt
package com.github.davinkevin.podcastserver.service

/**
 * Created by kevin on 02/09/2017
 */
//class ByteRangeResourceHandlerTest {
//
//    private val brh = ByteRangeResourceHandler()
//
//    @Test
//    fun `should expose a file resource`() {
//        /* GIVEN */
//        val file = Paths.get("/tmp/foo")
//        val request: HttpServletRequest = mock()
//        whenever(request.getAttribute(ByteRangeResourceHandler.ATTR_FILE)).thenReturn(file)
//
//        /* WHEN  */
//        val r = brh.getResource(request)
//
//        /* THEN  */
//        assertThat(r.file.toPath()).isEqualByComparingTo(file)
//    }
//
//    @Test
//    fun `should raise exception if no attribute`() {
//        /* GIVEN */
//        val request = mock(HttpServletRequest::class.java)
//        /* WHEN  */
//        assertThatThrownBy { brh.getResource(request) }
//                /* THEN  */
//                .hasMessage("Error during serving of byte range resources")
//    }
//
//}